recipes/glibmm/all/conanfile.py

```diff
@@ -43,129 +51,118 @@ def _giomm_lib(self):
         return f"giomm-{self._abi_version}"
 
     def validate(self):
-        if hasattr(self, "settings_build") and build.cross_building(self):
+        if hasattr(self, "settings_build") and cross_building(self):
             raise ConanInvalidConfiguration("Cross-building not implemented")
 
         if self.settings.compiler.get_safe("cppstd"):
             if self._abi_version == "2.68":
-                build.check_min_cppstd(self, 17)
+                check_min_cppstd(self, 17)
             else:
-                build.check_min_cppstd(self, 11)
+                check_min_cppstd(self, 11)
+
+        if is_msvc(self) and not self.options.shared:
+            raise ConanInvalidConfiguration("Static library build is not supported by MSVC")
 
         if self.options.shared and not self.options["glib"].shared:
             raise ConanInvalidConfiguration(
                 "Linking a shared library against static glib can cause unexpected behaviour."
             )
 
-        if self.options["glib"].shared and microsoft.is_msvc_static_runtime(self):
-            raise ConanInvalidConfiguration(
-                "Linking shared glib with the MSVC static runtime is not supported"
-            )
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+        if self.options["glib"].shared and is_msvc_static_runtime(self):
+            raise ConanInvalidConfiguration("Linking shared glib with the MSVC static runtime is not supported")
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def build_requirements(self):
-        self.build_requires("meson/0.64.1")
-        self.build_requires("pkgconf/1.9.3")
+        self.tool_requires("meson/1.0.0")
+        self.tool_requires("pkgconf/1.9.3")
 
     def requirements(self):
-        self.requires("glib/2.75.0")
-
+        self.requires("glib/2.75.2")
         if self._abi_version == "2.68":
             self.requires("libsigcpp/3.0.7")
         else:
             self.requires("libsigcpp/2.10.8")
 
+    def generate(self):
```

Comment:
generate() after source() please, otherwise it's hard to understand flow of execution and global logic of the recipe.